### PR TITLE
✨  [RUMF-794] Add isActive attribute to view events

### PR DESCRIPTION
**Pull request by webNeat**
## Motivation

Adds a new attribute `isActive` to view events, indicating if the view is currently active or not.

## Testing

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/master/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum/src/domain/rumEventsCollection/view/trackViews.spec.ts [hunk 1]
```diff
@@ -356,6 +356,46 @@ describe('rum track loading type', () => {
...
+    history.pushState({}, '', '/bar')
+    expect(getViewEvent(1).isActive).toBe(false)
+    expect(getViewEvent(2).isActive).toBe(true)
+  })
+
+  it('should collect view as active after a search change', () => {
+    setupBuilder.build()
+    history.pushState({}, '', '/foo?bar=qux')
+    expect(getViewEvent(1).isActive).toBe(true)
+  })
```
Comment (bcaudan) [inline packages/rum/src/domain/rumEventsCollection/view/trackViews.spec.ts:396]: what is interesting about search change?

Comment (webNeat) [reply packages/rum/src/domain/rumEventsCollection/view/trackViews.spec.ts:396]: it's a case when the location changes but the view remains active. The goal was to make sure that the view doesn't become inactive after any location change.

Comment (bcaudan) [reply packages/rum/src/domain/rumEventsCollection/view/trackViews.spec.ts:396]: ok, can you clarify you intent in the test description then?

File: packages/rum/src/domain/rumEventsCollection/view/trackViews.spec.ts [hunk 2]
```diff
@@ -356,6 +356,46 @@ describe('rum track loading type', () => {
...
+        lifeCycle.subscribe(LifeCycleEventType.VIEW_UPDATED, handler)
+        return trackViews(location, lifeCycle)
+      })
+  })
+
+  afterEach(() => {
+    setupBuilder.cleanup()
+  })
+
+  it('should collect initial view as active', () => {
```
Comment (bcaudan) [inline packages/rum/src/domain/rumEventsCollection/view/trackViews.spec.ts:380]:
```suggestion
  it('should set initial view as active', () => {
```

File: packages/rum/src/domain/rumEventsCollection/view/trackViews.spec.ts [hunk 3]
```diff
@@ -356,6 +356,46 @@ describe('rum track loading type', () => {
...
+    setupBuilder.build()
+    expect(getViewEvent(0).isActive).toBe(true)
+  })
+
+  it('should collect old view as inactive and new one as active after a route change', () => {
```
Comment (bcaudan) [inline packages/rum/src/domain/rumEventsCollection/view/trackViews.spec.ts:385]:
```suggestion
  it('should set old view as inactive and new one as active after a route change', () => {
```

File: packages/rum/src/domain/rumEventsCollection/view/trackViews.spec.ts [hunk 4]
```diff
@@ -356,6 +356,46 @@ describe('rum track loading type', () => {
...
+    setupBuilder.build()
+    history.pushState({}, '', '/bar')
+    expect(getViewEvent(1).isActive).toBe(false)
+    expect(getViewEvent(2).isActive).toBe(true)
+  })
+
+  it('should not collect view as inactive after any location change', () => {
```
Comment (bcaudan) [inline packages/rum/src/domain/rumEventsCollection/view/trackViews.spec.ts:392]:
```suggestion
  it('should keep view as active after a search change', () => {
```
FMU, any location change includes route change

File: packages/rum/src/domain/rumEventsCollection/view/trackViews.spec.ts [hunk 5]
```diff
@@ -356,6 +356,46 @@ describe('rum track loading type', () => {
...
+describe('rum track view is active', () => {
+  let setupBuilder: TestSetupBuilder
+  let handler: jasmine.Spy
+  let getViewEvent: (index: number) => View
+
+  beforeEach(() => {
+    ;({ handler, getViewEvent } = spyOnViews())
+
+    setupBuilder = setup()
+      .withFakeClock()
```
Comment (BenoitZugmeyer) [inline packages/rum/src/domain/rumEventsCollection/view/trackViews.spec.ts:368]: NIT: the fake clock is not used here, it could be removed

**Review by BenoitZugmeyer — APPROVED** [review]

**Review by bcaudan — APPROVED** [review]

**Pull request merged**